### PR TITLE
VH-82 Refactor thunks & merging data from different sources

### PR DESCRIPTION
**Pull request by rdgfuentes**
- [x] Single fetch to get vessel data from different sources
- [x] Resolve vessel data merge when received and hydrate VesselWithHistory object with vessel data (vessel class is not used anymore)
- [x] Unified Vessel type definition
- [x] Encapsulated vessel field history link and popup into the InfoFieldHistory component

### CODE REVIEW

File: applications/vessel-history/src/types/index.ts [hunk 1]
```diff
@@ -190,4 +136,6 @@ export type GFWDetail = {
   shipname: string
   source: string
   dataset: string
+  geartype: string
+  normalized_shipname: string
```
Comment (j8seangel) [inline applications/vessel-history/src/types/index.ts:140]: If we are using camelCase everywhere, why is this snake_case ?

File: applications/vessel-history/src/utils/async-slice.ts [hunk 2]
```diff
@@ -0,0 +1,156 @@
+import {
```
Comment (j8seangel) [inline applications/vessel-history/src/utils/async-slice.ts:1]: Did you find it useful ? Hope so!

**Review by j8seangel — APPROVED** [review]
I saw there is still a Profile.container.ts file which I'd remove as it is not used anymore.
In general it looks great! thanks for removing the class which makes it much easier to consume

File: applications/vessel-history/src/features/home/Home.tsx [hunk 3]
```diff
@@ -5,9 +5,9 @@ import { DebounceInput } from 'react-debounce-input'
 import Logo from '@globalfishingwatch/ui-components/dist/logo'
 import GFWAPI from '@globalfishingwatch/api-client'
 import { Spinner, IconButton } from '@globalfishingwatch/ui-components'
+import { VesselSearch } from '@globalfishingwatch/api-types/dist'
```
[the remark this answers is not in the document]
Comment (rdgfuentes) [reply applications/vessel-history/src/features/home/Home.tsx:8]: No idea, I wonder the same... I'll fix them

File: applications/vessel-history/src/types/index.ts (hunk 1 above)
Comment (rdgfuentes) [reply applications/vessel-history/src/types/index.ts:140]: To match the vessel API endpoint response 🤷‍♂️ , for some reason this field is returned with a different form.

File: applications/vessel-history/src/utils/async-slice.ts (hunk 2 above)
Comment (rdgfuentes) [reply applications/vessel-history/src/utils/async-slice.ts:1]: Completely useful, thanks!

File: applications/vessel-history/src/features/profile/components/InfoField.tsx [hunk 4]
```diff
@@ -1,42 +1,46 @@
...
+    value,
+    firstSeen: valuesHistory.slice().shift()?.endDate,
   }
   return (
     <div className={styles.identifierField}>
       <label>{label}</label>
-      <div onClick={() => displachOpenModal(field)}>
-        {field.value?.data}
-        <InfoFieldHistoric
+      <div onClick={() => displachOpenModal()}>
```
[the remark this answers is not in the document]
Comment (rdgfuentes) [reply applications/vessel-history/src/features/profile/components/InfoField.tsx]: Fixed

File: applications/vessel-history/src/features/profile/components/InfoFieldHistory.tsx [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+  return (
+    <Fragment>
+      <button className={styles.moreValues} onClick={toggleModalOpen}>{`+${history.length} ${t(
+        'common.previous',
+        'previous'
+      )} ${t(`common.${label}Plural`, `${label}s`)}`}</button>
+
+      {history.length && (
+        <Modal
+          title={label + ' History for ' + vesselName}
```
[the remark this answers is not in the document]
Comment (rdgfuentes) [reply applications/vessel-history/src/features/profile/components/InfoFieldHistory.tsx]: Fixed

**Pull request merged**